pkg/prometheus/promcfg.go

```diff
@@ -1289,6 +1289,14 @@ func (cg *ConfigGenerator) generateServiceMonitorConfig(
 		},
 	}...)
 
+	if ep.FilterRunning == nil || *ep.FilterRunning {
```

Comment:
Its not a blocker but perhaps its worth extracting the existing https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go#L708-L714 into its own function?